### PR TITLE
Add documentation for OpenShift

### PR DESCRIPTION
**Pull request by Emyrk**
Preview: https://codercom-em5j7gns3-codercom.vercel.app/docs/coder/v1.20/setup/kubernetes/openshift

### CODE REVIEW

**Comment by jawnsy** [comment]
[ch3540]

**Comment by jawnsy** [comment]
@Emyrk I can't tag you as reviewer since you opened the PR, please take a look at these changes, and then we'll ask Katie to review.

File: setup/kubernetes/openshift.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+This guide describes customizations to the OpenShift cluster as well as Coder
+that ensure an optimal user experience.
+
+[OpenShift Container Platform]: https://www.openshift.com/products/container-platform
+[Security Context Constraint]: https://docs.openshift.com/container-platform/4.7/authentication/managing-security-context-constraints.html
+
+## Prerequisites
+
+- An OpenShift cluster with a Project (Kubernetes namespace) for Coder
+- OpenShift command-line tools (`oc` and `kubectl`)
```
Comment (Emyrk) [inline setup/kubernetes/openshift.md:20]: Should we add Postgres here? Do we need to include docs on how to spin up postgres on openshift?

File: setup/kubernetes/openshift.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+Values file:
+
+```yaml
+coderd:
+  podSecurityContext:
+    runAsUser: null
+    seccompProfile: null
+  securityContext:
+    seccompProfile: null
+```
```
Comment (Emyrk) [inline setup/kubernetes/openshift.md:79]: Ah we will make them disable the runAsUser. Do we need to also null out the `runAsGroup`?

Comment (jawnsy) [reply setup/kubernetes/openshift.md:79]: Hmm, I didn't change runAsGroup, and deploying to our own OpenShift cluster seemed to work fine:

```
  Supplemental Groups Strategy: RunAsAny
    Ranges:				<none>
```

File: setup/kubernetes/openshift.md (hunk 1 above)
Comment (jawnsy) [reply setup/kubernetes/openshift.md:20]: This is a great point -- Our other guides don't mention PostgreSQL, I think we just expect users to find their own :thinking:

**Comment by jawnsy** [comment]
@khorne3 I think this is ready for your review now. This requires some changes to our Helm chart that are only available after 1.21 is released, so I'm not sure exactly how to handle this - maybe as a pull request into a release branch? Can you advise?

File: setup/kubernetes/openshift.md (hunk 1 above)
Comment (Emyrk) [reply setup/kubernetes/openshift.md:20]: I am ok with that personally, if anyone ends up needing those docs, I'm sure we will hear about it. 

Maybe we should mention it in the prereqs though?

File: setup/kubernetes/openshift.md (hunk 2 above)
Comment (Emyrk) [reply setup/kubernetes/openshift.md:79]: I guess that answers that then

**Comment by khorne3** [comment]
@jawnsy I started a 1.21 branch so we can merge this into that, instead of main, when ready

**Comment by khorne3** [comment]
@Emyrk just some clarification questions: to use Openshift, they should:

1. Modify pod and container security contexts
2. Choose between Option 1 (add service account) and Option 2 (build images)

where both steps are required?

**Comment by jawnsy** [comment]
@khorne3 Yup, your understanding is correct!

I've re-targeted this change to the release branch and re-based, so everything should be ready to go now

**Review by khorne3 — APPROVED** [review]

File: setup/kubernetes/openshift.md (hunk 1 above)
Comment (jawnsy) [reply setup/kubernetes/openshift.md]: Well, if we mention it for OpenShift, we should mention it for Azure/EKS/GKE, too, for consistency... and then I suppose it's best to do that as a separate PR

**Pull request merged**